src/build/env.py

```diff
@@ -137,78 +144,58 @@ def install(self, requirements):  # type: (Iterable[str]) -> None
             os.unlink(req_file.name)
 
 
-if sys.version_info[0] == 2:  # noqa: C901 # disable if too complex
+def _create_isolated_env_virtualenv(path):  # type: (str) -> Tuple[str, str]
+    """
+    On Python 2 we use the virtualenv package to provision a virtual environment.
 
-    def _create_isolated_env(path):  # type: (str) -> Tuple[str, str]
-        """
-        On Python 2 we use the virtualenv package to provision a virtual environment.
+    :param path: the folder where to create the isolated build environment
+    :return: the isolated build environment executable, and the pip to use to install packages into it
+    """
+    cmd = [str(path), '--no-setuptools', '--no-wheel', '--activators', '']
+    result = virtualenv.cli_run(cmd, setup_logging=False)
+    executable = str(result.creator.exe)
+    script_dir = str(result.creator.script_dir)
+    return executable, script_dir
 
-        :param path: the folder where to create the isolated build environment
-        :return: the isolated build environment executable, and the pip to use to install packages into it
-        """
-        from virtualenv import cli_run
 
-        cmd = [str(path), '--no-setuptools', '--no-wheel', '--activators', '']
-        if pip is not None:
-            cmd.append('--no-pip')
-        result = cli_run(cmd, setup_logging=False)
-        executable = str(result.creator.exe)
-        pip_executable = executable if pip is None else sys.executable
-        return executable, pip_executable
+def _create_isolated_env_venv(path):  # type: (str) -> Tuple[str, str]
+    """
+    On Python 3 we use the venv package from the standard library, and if host python has no pip the ensurepip
+    package to provision one into the created virtual environment.
 
+    :param path: the folder where to create the isolated build environment
+    :return: the isolated build environment executable, and the pip to use to install packages into it
+    """
+    import venv
 
-else:
+    venv.EnvBuilder(with_pip=True).create(path)
+    executable, script_dir = _find_executable_and_scripts(path)
+    # avoid the setuptools from ensurepip to break the isolation
+    if sys.version_info[0:2] == (3, 5):  # python3.5 by default comes with pip 9 that's too old, for new standards
+        subprocess.check_call([executable, '-m', 'pip', 'install', '-U', 'pip'])
```

Comment:
These commands used to be run in isolated mode (`-I`), why is that not done anymore?

Reply:
Because beforehand the pip could have been global, now it's guaranteed to be within a virtual environment so it's no longer needed.

Reply:
`-I` still provides isolation from (a) the cwd and (b) host Python env vars.